network/simple-onchain-discovery/src/lib.rs

```diff
@@ -0,0 +1,168 @@
+            "Histogram of busy time of spent in event processing loop"
+        )
+        .unwrap(),
+    )
+});
+
+pub struct ConfigurationChangeListener {
+    conn_mgr_reqs_tx: channel::Sender<ConnectivityRequest>,
+    role: RoleType,
+}
+
+fn extract_validator_updates(
+    validator_set: ValidatorSet,
+) -> (
+    Vec<ConnectivityRequest::UpdateAddresses>,
+    ConnectivityRequest::UpdateEligibleNodes,
+) {
+    let validator_keys = validator_set.payload().to_vec();
+    let nodes = ConnectivityRequest::UpdateEligibleNodes(
+        validator_keys
+            .into_iter()
+            .map(|validator| {
+                (
+                    *validator.account_address(),
+                    NetworkPublicKeys {
+                        identity_public_key: validator.network_identity_public_key(),
+                        signing_public_key: validator.network_signing_public_key().clone(),
+                    },
+                )
+            })
+            .collect(),
+    );
+    let addresses: Vec<ConnectivityRequest::UpdateAddresses()> = validator_keys
+        .into_iter()
+        .map(|validator| {
+            ConnectivityRequest::UpdateAddresses(
+                DiscoverySource::OnChain,
+                validator.account_address().clone(),
+                validator.config().validator_network_address(),
+            )
+        })
+        .collect();
+    (addresses, nodes)
+}
+
+fn extract_full_node_updates(
+    full_node_set: ValidatorSet,
+) -> (
+    Vec<ConnectivityRequest::UpdateAddresses>,
+    ConnectivityRequest::UpdateEligibleNodes,
+) {
+    let full_node_infos = full_node_set.payload().to_vec();
+    let nodes = ConnectivityRequest::UpdateEligibleNodes(
+        full_node_infos
+            .into_iter()
+            .map(|full_node| {
+                (
+                    // TODO:  This seems like it is wrong and should be something else.
+                    *full_node.account_address().clone(),
+                    NetworkPublicKeys {
+                        identity_public_key: full_node.config().full_node_network_identity_public_key().clone(),
+                        // TODO:  there are not separate keys for the full_node.  Are they therefore the same?
+                        signing_public_key: full_node.config().full_node_network_identity_public_key().clone(),
+                    },
+                )
+            })
+            .collect(),
+    );
+    let addresses: Vec<ConnectivityRequest::UpdateAddresses()> = full_node_infos
+        .into_iter()
+        .map(|full_node| {
+            ConnectivityRequest::UpdateAddresses(
+                DiscoverySource::OnChain,
+                // TODO:  again, this seems wrong for the account address
+                full_node.account_address().clone(),
+                full_node.config().full_node_network_address().clone(),
+            )
+        })
+        .collect();
+    (addresses, nodes)
+}
```

Comment:
@davidiw helper function to extract the full_node relevant information.

There are a number of pieces that are very opaque here and feel like they might be wrong/misguided/incomplete.   

Key issues:
1). what is the account_address for a full_node?  Is it the same as for the validator on that machine?  (i.e., is the account_address actually a machine address????)
2).  are the full_node identity and signing keys guaranteed to be the same?